tests/test_lane.py

```diff
@@ -3264,41 +3264,43 @@ def _w(**kwargs):
-        lq_litfic.quality = 0
-        hq_sf = _w(title="HQ SF", genre="Science Fiction", fiction=True)
-        hq_sf.random = 0.25
+        self.hq_litfic = _w(title="HQ LitFic", fiction=True, genre='Literary Fiction')
+        self.hq_litfic.quality = 0.8
+        self.lq_litfic = _w(title="LQ LitFic", fiction=True, genre='Literary Fiction')
+        self.lq_litfic.quality = 0
+        self.hq_sf = _w(title="HQ SF", genre="Science Fiction", fiction=True)
 
         # Add a lot of irrelevant genres to one of the works. This
         # will clutter up the materialized view, but it won't affect
         # the results.
         for genre in ['Westerns', 'Horror', 'Erotica']:
             genre_obj, is_new = Genre.lookup(self._db, genre)
-            get_one_or_create(self._db, WorkGenre, work=hq_sf, genre=genre_obj)
-
-        hq_sf.quality = 0.8
-        mq_sf = _w(title="MQ SF", genre="Science Fiction", fiction=True)
-        mq_sf.quality = 0.6
-        lq_sf = _w(title="LQ SF", genre="Science Fiction", fiction=True)
-        lq_sf.quality = 0.1
-        hq_ro = _w(title="HQ Romance", genre="Romance", fiction=True)
-        hq_ro.quality = 0.8
-        hq_ro.random = 0.75
-        mq_ro = _w(title="MQ Romance", genre="Romance", fiction=True)
-        mq_ro.quality = 0.6
-        lq_ro = _w(title="LQ Romance", genre="Romance", fiction=True)
-        lq_ro.quality = 0.1
-        nonfiction = _w(title="Nonfiction", fiction=False)
+            get_one_or_create(self._db, WorkGenre, work=self.hq_sf, genre=genre_obj)
+
+        self.hq_sf.quality = 0.8
+        self.mq_sf = _w(title="MQ SF", genre="Science Fiction", fiction=True)
+        self.mq_sf.quality = 0.6
+        self.lq_sf = _w(title="LQ SF", genre="Science Fiction", fiction=True)
+        self.lq_sf.quality = 0.1
+        self.hq_ro = _w(title="HQ Romance", genre="Romance", fiction=True)
+        self.hq_ro.quality = 0.8
+        self.mq_ro = _w(title="MQ Romance", genre="Romance", fiction=True)
+        self.mq_ro.quality = 0.6
+        # This work is in a different language -- necessary to run the
+        # LQRomanceEntryPoint test below.
+        self.lq_ro = _w(title="LQ Romance", genre="Romance", fiction=True, language='lan')
+        self.lq_ro.quality = 0.1
+        self.nonfiction = _w(title="Nonfiction", fiction=False)
 
         # One of these works (mq_sf) is a best-seller and also a staff
         # pick.
-        best_seller_list, ignore = self._customlist(num_entries=0)
-        best_seller_list.add_entry(mq_sf)
+        self.best_seller_list, ignore = self._customlist(num_entries=0)
+        self.best_seller_list.add_entry(self.mq_sf)
 
-        staff_picks_list, ignore = self._customlist(num_entries=0)
-        staff_picks_list.add_entry(mq_sf)
+        self.staff_picks_list, ignore = self._customlist(num_entries=0)
+        self.staff_picks_list.add_entry(self.mq_sf)
+
+    def test_groups(self):
```

Comment:
Now that this is a search test should it check self.search to decide whether to run?

Reply:
Or, maybe it's important enough now that we should always require elasticsearch to run the tests.